Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/CartesianYeeAlgorithm.H

```diff
@@ -145,14 +147,15 @@ struct CartesianYeeAlgorithm {
     static amrex::Real DownwardDz (
         amrex::Array4<amrex::Real> const& F,
         amrex::Real const * const coefs_z, int const /*n_coefs_z*/,
-        int const i, int const j, int const k, int const ncomp=0 ) {
+        int const i, int const j, int const k, int const ncomp=0,
+        amrex::Real const mu_up =1.0, amrex::Real const mu_down = 1.0  ) {
 
         using namespace amrex;
         Real const inv_dz = coefs_z[0];
 #if defined WARPX_DIM_3D
-        return inv_dz*( F(i,j,k,ncomp) - F(i,j,k-1,ncomp) );
+        return inv_dz*( F(i,j,k,ncomp)/mu_up - F(i,j,k-1,ncomp)/mu_down );
```

Comment:
Instead of passing in `mu_up` and `mu_down`, you could pass in the `mu` array and do the indexing in this routine, i.e. `return inv_dz*( F(i,j,k,ncomp)/mu(i,j,k) - F(i,j,k-1,ncomp)/mu(i,j,k-1) );`. This wold avoid having the ifdefs in the calling code. Though if in the future you will have mu be purely analytic, this might not be any better.

A detail is that this routine is also used when differencing E so it might be a little confusing that `mu` is used here.

Reply:
Oh yes. I can pass the mu_array as well. But for the vacuum case, the default mu_array would be nullptr. And then I might have to put in if conditions for the array. 

I just saw the next comment. I agree with that! I can definitely do a parallelFor and first divide B/mu, then do the curlB operations using DownwardD, and then multiply mu back, to regain B.

Thank you for this idea @dpgrote !